src/fmt/writer/termcolor/shim_impl.rs

```diff
@@ -42,10 +32,9 @@ impl BufferWriter {
         // instead of using the streams directly.
         // This is so their output can be captured by `cargo test`.
         match &self.target {
-            // Safety: If the target type is `Pipe`, `target_pipe` will always be non-empty.
-            WritableTarget::Pipe(pipe) => pipe.lock().unwrap().write_all(&buf.0)?,
             WritableTarget::Stdout => print!("{}", String::from_utf8_lossy(&buf.0)),
             WritableTarget::Stderr => eprint!("{}", String::from_utf8_lossy(&buf.0)),
+            WritableTarget::Pipe(_) => unreachable!(),
```

Comment:
Maybe I am missing something quite obvious, but why is this unreachable?

Reply:
`BufferWriter::pipe` was removed, so there is only `BufferWriter::stdout` and `BufferWriter::stderr` left to create a `BufferWriter` and both of those can't have the Pipe Target.